micro_ros_setup/config/freertos/esp32/create.sh

```diff
@@ -0,0 +1,38 @@
+pushd $FW_TARGETDIR >/dev/null
+    # Install toolchain
+    mkdir toolchain
+
+    # install prerequesites for esp-idf
+    # sudo apt-get install -y gperf python3-venv
```

Comment:
I'm not sure if I should run this `sudo` command here to install apt dependencies.
Or is it better to just check if the dependencies are satisfied and if not request the user to install them themself?

Reply:
are they available using rosdep?

Reply:
yes they are should I put them into package.xml?

Reply:
yes, they should be in the `package.xml` inside `micro-ros-build/micro_ros_setup/config/freertos/esp32/`. This way, the build system should resolve this dependecies.

Also, can you add this platform to the CI.yml?

Reply:
Ok I will.

Reply:
Done :)
The CI step for esp32 still fails because micro-ROS/freertos_apps/pull/12 is not merged to foxy.